### ISSUE TITLE
Difference between helmsman and helmfile

### ISSUE THREAD

**Issue by eicnix** [Praqma/helmsman#18]
How does helmsman differentiate itself versus [helmfile](https://github.com/roboll/helmfile)? 

Both try to solve managing multiple helm applications from a single configuration file.

**Comment by sami-alajrami** [comment]
@eicnix Thanks for your question.

Actually, when we created helmsman some months ago, we were not aware of helmfile. But when we got to know about it, we realized that despite sharing the same general goal, there were major differences in the details. For example:

-  We use TOML format for describing desired states. This is a more human-readable format than yaml and is not whitespace sensitive.
- Helmsman desired state file is portable (i.e. it does not require helm/kubectl to be configured. It can do that for you.) 
    - Helmsman can read secrets for connecting to your clusters from S3 & GCS buckets.
    - Can setup your helm repositories for you.
    - Can define k8s namespaces (and creates them if they don't exist) 
- Helmsman provides namespace and release protection against accidental desired state file commits.
- Helmsman allows you to delete(with/without purge)/install/rollback/test/upgrade/move(across namespaces) releases based on your desired state changes. 
- Helmsman works with private helm repositories hosted in S3 or GCS using helm plugins (only needs AWS or GCS env variables set)

For more details, have a look at our [documentation](https://github.com/Praqma/helmsman/tree/master/docs)
